File: templates/java/spring/spring_service.java
#!target:spring/src/main/java/{{.global.pkg}}/service/{{.table.Title}}Service.java
package {{pkg "java" .global.pkg}}.service

import {{pkg "java" .global.pkg}}.pojo.{{.table.Title}}Entity
import {{pkg "java" .global.pkg}}.repo.{{.table.Title}}JpaRepository
import org.springframework.stereotype.Service
import org.springframework.data.repository.findByIdOrNull
import java.util.*
import javax.annotation.Resource
{{$tableTitle := .table.Title}}
{{$pkType := type "java" .table.PkType}}
/** {{.table.Comment}}服务  */
@Service
public class {{.table.Title}}Service {
    @Resource
    private {{$tableTitle}}JpaRepository repo;


    /** 查找{{.table.Comment}} */
    public {{$tableTitle}}Entity findByIdOrNull({{$pkType}} id){
        return this.repo.findByIdOrNull(id);
    }

    /** 保存{{.table.Comment}} */
    public {{$tableTitle}}Entity save{{$tableTitle}}({{$tableTitle}}Entity {{.table.Name}}){
        return this.repo.save({{.table.Name}})
    }


    /** 批量保存{{.table.Comment}} */
    public Iterable<{{$tableTitle}}Entity> saveAll{{$tableTitle}}(Iterable<{{$tableTitle}}Entity> entities){
        return this.repo.saveAll(entities);
    }

    /** 删除{{.table.Comment}} */
    public void deleteById({{$pkType}} id) {
        this.repo.deleteById(id);
    }

}

File: templates/java/spring/spring_service.kt
#!target:spring/src/main/kotlin/{{.global.pkg}}/service/{{.table.Title}}Service.kt
package {{pkg "java" .global.pkg}}.service

import {{pkg "java" .global.pkg}}.pojo.{{.table.Title}}Entity
import {{pkg "java" .global.pkg}}.repo.{{.table.Title}}JpaRepository
import org.springframework.stereotype.Service
import org.springframework.data.repository.findByIdOrNull
import net.fze.util.catch
import javax.annotation.Resource
{{$tableTitle := .table.Title}}
{{$pkProp := lower_title .table.Pk}}
{{$pkType := type "kotlin" .table.PkType}}
/** {{.table.Comment}}服务  */
@Service
class {{.table.Title}}Service {

    @Resource
    lateinit var repo: {{$tableTitle}}JpaRepository

    // 根据ID查找{{.table.Comment}}
    fun findByIdOrNull(id:{{$pkType}}):{{$tableTitle}}Entity?{
        return this.repo.findByIdOrNull(id)
    }

    // 保存{{.table.Comment}}
    fun save{{$tableTitle}}(e: {{$tableTitle}}Entity):Error? {
        return catch {
            var dst: {{$tableTitle}}Entity
            if (e.{{$pkProp}} > 0) {
                dst = this.repo.findByIdOrNull(e.{{$pkProp}})!!
            } else {
                dst = {{$tableTitle}}Entity()
            }
            {{range $i,$c := .columns}}
            dst.{{lower_title $c.Prop}} = e.{{lower_title $c.Prop}}{{end}}
            this.repo.save(dst)
            null
        }.error()
    }

    // 批量保存{{.table.Comment}}
    fun saveAll{{$tableTitle}}(entities:Iterable<{{$tableTitle}}Entity>): Iterable<{{$tableTitle}}Entity>{
        return this.repo.saveAll(entities)
    }

    // 删除{{.table.Comment}}
    fun deleteById(id:{{$pkType}}) {
         this.repo.deleteById(id)
    }

}
